core/src/main/java/org/infinispan/protostream/TagHandler.java

```diff
@@ -13,8 +15,10 @@
 
    /**
     * Start of top level message. Do whatever required init here.
+    *
+    * @param descriptor The {@link GenericDescriptor} for the top level Enum or Message. For primitives it is null.
     */
-   default void onStart() {
+   default void onStart(AnnotatedDescriptor descriptor) {
```

Comment:
> I'd prefer the descriptor arg to be a GenericDescriptor

np, I need an broader descriptor for the cases where there is no message being rendered